### PR TITLE
update Dockerfile

### PR DESCRIPTION
**Pull request by Remi-Gau**
Adapted from the spm repo

### TODO
- [ ] set up github action to build and push automatically to docker hub (https://github.com/marketplace/actions/build-and-push-docker-images)
- [ ] update doc
- [ ] make recipe download the submodules

---

### issues

- [ ] can the pipeline be run if launched interactively
- [ ] look into BIDS apps to see how to standardize the docker image
- [ ] make it more ligthweight
- [ ] see linked issues for more details (create a matlab image)

### CODE REVIEW

**Comment by marcobarilari** [comment]
I can help on point 1 cause I want to understand how to set up an action

File: docker/Dockerfile [hunk 1]
```diff
@@ -30,11 +28,10 @@ RUN mkdir /opt/spm12 \
 RUN octave --no-gui --eval "addpath(\"/opt/spm12\"); savepath ();"
 
 # get the CPP BIDS pipeline code
-RUN mkdir /CPP_BIDS_pipeline /output /code \
- && git clone https://github.com/cpp-lln-lab/CPP_BIDS_SPM_pipeline.git /CPP_BIDS_pipeline/
-
-RUN octave --no-gui --eval "addpath(genpath(\"/CPP_BIDS_pipeline\")); savepath ();"
+RUN mkdir /CPP_SPM /output /code \
+ && git clone https://github.com/cpp-lln-lab/CPP_SPM.git /CPP_SPM/
```
Comment (marcobarilari) [inline docker/Dockerfile:32]: so, on this line, it clones the whole repo, right? Would not be better to make it fix to a release instead?

Comment (Remi-Gau) [reply docker/Dockerfile:32]: yes indeed! 

and actually make the clone recursive so we get the submodule

a possibility to get a specific release would be to just `git checkout` the tag of that realease.

**Comment by marcobarilari** [comment]
also related to point (1), have you checked this? https://docs.docker.com/docker-hub/builds/

**Comment by Remi-Gau** [comment]
> also related to point (1), have you checked this? https://docs.docker.com/docker-hub/builds/

Not yet. :wink:

**Comment by marcobarilari** [comment]
I am pushing the image in our organization on the docker hub

**Comment by Remi-Gau** [comment]
https://www.youtube.com/watch?v=Ps0MfBG5-Uo

**Comment by marcobarilari** [comment]
the image is on the hub and it can be run with 

```bash
docker run -it --rm \
-v [output_folder]:/output \
-v [code_folder]:/code cpplab/cpp_spm:0.1.0
```

this will be to be updated in the docs and also to remind that there no need to build it locally (though I would take that part)

File: docs/source/docker.rst [hunk 2]
```diff
@@ -1,35 +1,37 @@
...
-    # running the command 'ls' should show MoAEpilot_run.m and the content of 
+    # to change to the code folder inside the container
+    # running the command 'ls' should show MoAEpilot_run.m and the content of
     # the folder
-    cd demos/MoAE 
+    cd demos/MoAE
 
     # To run the batch_download_run script
-    octave --no-gui --eval MoAEpilot_run 
+    octave --no-gui --eval MoAEpilot_run
```
Comment (Remi-Gau) [inline docs/source/docker.rst:23]: That's when I tried this part I realize that I realize that the image was built but did not include the sub modules .

It failed miserably

**Comment by Remi-Gau** [comment]
closed temporarily as the the commits have been cherry picked onto PR #367

**Comment by Remi-Gau** [comment]
> also related to point (1), have you checked this? https://docs.docker.com/docker-hub/builds/

@marcobarilari 
I checked the details of this and it seems that automatic build of images on push are only available if we pay for docker hub.
Creating some scripts to help with the image building and we will try to automate this in CI later.

**Pull request closed without merging**